lib/improver/tests/ensemble_calibration/ensemble_calibration/test_EstimateCoefficientsForEnsembleCalibration.py

```diff
@@ -171,6 +172,150 @@ def test_statsmodels_realizations(self, warning_list=None):
+    @ManageWarnings(
+        ignored_messages=IGNORED_MESSAGES, warning_types=WARNING_TYPES)
+    def test_coefficients_from_mean(self):
+        """Test that the expected coefficient cube is returned when the
+        ensemble mean is used as the predictor."""
+        expected_coeff_names = ["gamma", "delta", "a", "beta"]
+        result = self.plugin.create_coefficients_cube(
+            self.optimised_coeffs, self.current_forecast)
+        self.assertEqual(result, self.expected)
+        self.assertEqual(
+            self.plugin.coeff_names, expected_coeff_names)
+
+    @ManageWarnings(
+        ignored_messages=IGNORED_MESSAGES, warning_types=WARNING_TYPES)
+    def test_coefficients_from_realizations(self):
+        """Test that the expected coefficient cube is returned when the
+        ensemble realizations are used as the predictor."""
+        expected_coeff_names = (
+            ["gamma", "delta", "a", "beta0", "beta1", "beta2"])
+        predictor_of_mean_flag = "realizations"
+        optimised_coeffs = [0, 1, 2, 3, 4, 5]
+
+        # Set up an expected cube.
+        coefficient_index = iris.coords.DimCoord(
+            optimised_coeffs, long_name="coefficient_index", units="1")
+        dim_coords_and_dims = [(coefficient_index, 0)]
+
+        coefficient_name = iris.coords.AuxCoord(
+            expected_coeff_names, long_name="coefficient_name",
+            units="no_unit")
+
+        aux_coords_and_dims = [
+            (coefficient_name, 0),
+            (self.current_forecast.coord("time"), None),
+            (self.current_forecast.coord("forecast_reference_time"), None),
+            (self.current_forecast.coord("forecast_period"), None)]
+
+        attributes = {"mosg__model_configuration": "uk_det",
+                      "diagnostic_standard_name": "air_temperature"}
+
+        expected = iris.cube.Cube(
+            optimised_coeffs, long_name="emos_coefficients", units="1",
+            dim_coords_and_dims=dim_coords_and_dims,
+            aux_coords_and_dims=aux_coords_and_dims, attributes=attributes)
+
+        plugin = Plugin(distribution=self.distribution,
+                        desired_units=self.desired_units,
+                        predictor_of_mean_flag=predictor_of_mean_flag)
+        result = plugin.create_coefficients_cube(
+            optimised_coeffs, self.current_forecast_with_realizations)
+        self.assertEqual(result, expected)
+        self.assertArrayEqual(
+            result.coord("coefficient_name").points, expected_coeff_names)
+
+    @ManageWarnings(
+        ignored_messages=IGNORED_MESSAGES, warning_types=WARNING_TYPES)
+    def test_forecast_period_coordinate_not_present(self):
+        """Test that the coefficients cube is created correctly when the
+        forecast_period coordinate is not present within the input cube."""
+        expected = self.expected.copy()
+        expected.remove_coord("forecast_period")
+        self.current_forecast.remove_coord("forecast_period")
```

Comment:
Why do you delete the coordinate in place for the forecast cube but on a copy for the expected result? Can they not both be done in place saving an entire line (a whole line!) 😄

Reply:
Done.